language-support/java/bindings/src/main/java/com/daml/ledger/javaapi/data/DamlList.java

```diff
@@ -33,6 +32,18 @@ public DamlList(@NonNull Value...values) {
         return values;
     }
 
+    public static <T> Collector<T, ArrayList<Value>, DamlList> collector(Function<T, Value> valueMapper) {
+        return Collector.of(
+                ArrayList::new,
+                (acc, entry) -> acc.add(valueMapper.apply(entry)),
+                (left, right) -> {
+                    left.addAll(right);
+                    return left;
+                },
+                DamlList::new
+        );
```

Comment:
Again, we can use:

```java
return collectingAndThen(Collectors.toList(), DamlList::new);
```

Reply:
This is not so simple. I have to map the list with `valueMapper`. Not sure it is worthwhile to make that following your approach. 

Do you prefer I drop the collector for List ? It is true that is not very usefull. I put it to here to follow the same pattern as Maps.

Reply:
Makes sense. You can use `Collectors.mapping` but it’s up to you whether you find this cleaner or not.